network/src/network.rs

```diff
@@ -1261,6 +1261,69 @@ impl NetworkController {
     }
 
+    /// disconnect peers with matched peer_ip or peer_ip_network, eg: 192.168.0.2 or 192.168.0.0/24
+    pub fn disconnect_peer_in_ip_rang(
+        &self,
+        address: IpNetwork,
+        duration: Duration,
+        reason: String,
+    ) {
+        match address {
+            IpNetwork::V4(match_ipn) => {
+                self.network_state.with_peer_registry(|reg| {
+                    for (peer_index, peer) in reg.peers().iter() {
+                        // assume compare first protocol ip4, ignore the other protocols
+                        if let Protocol::Ip4(peer_ip) =
+                            peer.connected_addr.iter().next().expect("first protocol")
+                        {
+                            let peer_ipn =
+                                Ipv4Network::new(peer_ip, 32).expect("Ipv4Network from ipv4Addr");
+                            if peer_ipn.is_subnet_of(match_ipn) {
+                                let message = format!(
+                                    "Ban peer {:?} for {} seconds, reason: {}",
+                                    peer_ipn,
+                                    duration.as_secs(),
+                                    reason
+                                );
+                                let _ = disconnect_with_message(
+                                    &self.p2p_control,
+                                    *peer_index,
+                                    &message,
+                                );
+                            }
+                        }
+                    }
+                });
+            }
+            IpNetwork::V6(match_ipn) => {
+                self.network_state.with_peer_registry(|reg| {
+                    for (peer_index, peer) in reg.peers().iter() {
+                        // assume compare first protocol ip6, ignore the other protocols
+                        if let Protocol::Ip6(peer_ip) =
+                            peer.connected_addr.iter().next().expect("first protocol")
+                        {
+                            let peer_ipn =
+                                Ipv6Network::new(peer_ip, 32).expect("Ipv6Network from ipv6Addr");
+                            if peer_ipn.is_subnet_of(match_ipn) {
+                                let message = format!(
+                                    "Ban peer {:?} for {} seconds, reason: {}",
+                                    peer_ipn,
+                                    duration.as_secs(),
+                                    reason
+                                );
+                                let _ = disconnect_with_message(
+                                    &self.p2p_control,
+                                    *peer_index,
+                                    &message,
+                                );
+                            }
+                        }
+                    }
+                });
+            }
+        }
```

Comment:
please use some built-in fn:

```suggestion
       self.network_state.with_peer_registry(|reg| {
            reg.peers().iter().for_each(|(peer_index, peer)| {
                multiaddr_to_socketaddr(peer.connected_addr).map(|addr| {
                    if address.contains(addr.ip()) {
                        self.ban_peer(peer_index, duration, reason));
                    }
                });
            })
        });
```

Reply:
the built-in fn codes are awesome, I keep scratching head until meet it.

Reply:
done! 
- i attach message instead of  reason
- clippy suggest to use 
 `if Some(..) =  multiaddr_to_socketaddr(peer.connected_addr)` cause of this [reason](https://rust-lang.github.io/rust-clippy/master/index.html#option_map_unit_fn)